doc/ecmp/wecmp_hld.md

```diff
@@ -0,0 +1,174 @@
+```
+  /* Handler to process ECMP Weight information rtnl messages */
+  string RouteSync::getNextHopWt(struct rtnl_route *route_obj);
+```
+##### RouteOrch
+RouteOrch is an existing component of the OrchAgent daemon in the SWSS container.  RouteOrch monitors operations on Route related tables in APPL DB and converts those operations in SAI commands to manage Route entries.  Additionally RouteOrch coordinates Next Hop object operations with NeighOrch and converts operations into SAI commands to manage Next Hop Group objects.
+
+###### routeorch functions
+The following functions are modified to support weight field in Next Hop Group Members.
+```
+  /* Handler to process ECMP Weight information from APPL DB */
+  void RouteOrch::doPrefixTask(Consumer& consumer)
+  /* Handler to call SAI API with Weight information */
+  bool RouteOrch::validnexthopinNextHopGroup(const NextHopKey &nexthop)
+```
+##### NextHopKey
+NextHopKey is an existing structure of the OrchAgent daemon in the SWSS container.  NextHopKey is used by both RouteOrch and NeighOrch to coordinate Next Hop operations.
+NextHopKey class is modified to include next hop weight field.
+```
+struct NextHopKey
+{
+    IpAddress           ip_address;     // neighbor IP address
+    string              alias;          // incoming interface alias
+    uint8_t             weight;         // NH weight for NHGs
+    ... struct definition abbreviated ...
+};
+```
+
+### SAI API
+This section describes SAI APIs used to support weighted ECMP routes.
+
+Full details about SAI Next Hop Group API and attributes are described here:
+https://github.com/opencomputeproject/SAI/blob/master/inc/sainexthopgroup.h
+To facilitate weighted ECMP route functionality, the following update was made to the SAI Next Hop Group API:
+```
+    /**
+     * @brief Member weights
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 1
+     */
+    SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT,
+```
+
+### Configuration and management
+There is no Configuration/management change for this feature.
+
+#### CLI Enhancements
+There is no CLI change for this feature. Weight information in ECMP is shown in ip route show command.
+
+#### Config DB Enhancements
+There is no config DB change.
+
+### Warmboot and Fastboot Design Impact
+- Weighted ECMP functionality continues across warm reboot.
+- Support for planned system warm restart.
+- Support for SWSS docker warm restart.
+
+### Restrictions/Limitations
+The routing stack configuration/process sends weight value in netlink messages is not handled in this document.
+Example configuartion in FRR might be http://docs.frrouting.org/en/latest/bgp.html#weighted-ecmp-using-bgp-link-bandwidth.
+
+### Testing Requirements/Design
+Testing of weighted ECMP can be done either by running routing stack(FRR with BGP), or use ip route command to set weight value in next hop group members.
+
+#### Unit Test cases
+
+#### System Test cases

```

Comment:
please describe system test cases.

Reply:
Will add: 
set up ECMP route with weight through config_db static_route table(e.g. two routes in NHG with weight of 2,8).
Send traffic with enough tuple variations and observe traffic load unevenly distributed on the two routes in the NHG as (20% and 80%).